src/blueprint_provider/default_vm_blueprint_provider.cpp

```diff
@@ -135,14 +208,38 @@ mp::Query mp::DefaultVMBlueprintProvider::fetch_blueprint_for(const std::string&
         }
     }
 
-    auto blueprint_instance = blueprint_config["instances"][blueprint_name];
+    auto blueprint_instance = blueprint_config[instances_key][blueprint_name];
 
     // TODO: Abstract all of the following YAML schema boilerplate
-    if (blueprint_instance["image"])
+
+    if (blueprint_config["blueprint-version"].as<std::string>() == "v2")
```

Comment:
I think instead of hard coding `v2` here, it would be better to leverage the constant in `blueprint_dir_versions`.

Reply:
That's a good idea. I made the strings `"v1"` and `"v2"` `constexpr`. Is that what you mean?

Reply:
Yeah, pretty much:)